test/test_foreach.py

```diff
@@ -157,90 +126,27 @@ def _test_bin_op_list_alpha(self, device, dtype, foreach_op, foreach_op_, torch_
-                            res = fe_op(tensors1)
-                        break
-
-                if dtype in [torch.complex64, torch.complex128] and not support_complex:
-                    if not (self.device_type == 'cpu' and torch_op in [torch.sigmoid]):
-                        # not using assertRaisesRegex due to different error messages
-                        with self.assertRaises(RuntimeError):
-                            expected = [torch_op(tensors1[i]) for i in range(N)]
-
-                        with self.assertRaises(RuntimeError):
-                            res = fe_op(tensors1)
-                        break
-
-                expected = [torch_op(tensors1[i].to(dtype=control_dtype)).to(dtype=dtype) for i in range(N)]
-                res = fe_op(tensors1)
-                if (dtype is torch.float16 or dtype is torch.bfloat16) and TEST_WITH_ROCM:
-                    self.assertEqual(res, expected, atol=1.e-3, rtol=self.dtype_precisions[dtype][0])
-
-                    fe_op_(tensors1)
-                    self.assertEqual(res, tensors1)
-                else:
-                    self.assertEqual(res, expected)
-
-                    fe_op_(tensors1)
-                    self.assertEqual(res, tensors1)
-
-    # Separate test for abs due to a lot of special cases
-    # Absolute value of a complex number a + bj is defined as sqrt(a^2 + b^2), i.e. a floating point
-    @dtypes(*torch.testing.get_all_dtypes())
-    def test_abs(self, device, dtype):
+    @ops(foreach_unary_op_db)
+    def test_unary(self, device, dtype, op):
         for N in N_values:
-            tensors1 = self._get_test_data(device, dtype, N)
-            # Mimics cuda kernel dtype flow.  With fp16/bf16 input, runs in fp32 and casts output back to fp16/bf16.
-            control_dtype = torch.float32 if (self.device_type == 'cuda' and
-                                              (dtype is torch.float16 or dtype is torch.bfloat16)) else dtype
-
-            if dtype == torch.bool and self.device_type == 'cpu':
-                with self.assertRaisesRegex(RuntimeError, r"not implemented for"):
-                    expected = [torch.abs(tensors1[i].to(dtype=control_dtype)).to(dtype=dtype) for i in range(N)]
-                continue
-
-            expected = [torch.abs(tensors1[i].to(dtype=control_dtype)).to(dtype=dtype) for i in range(N)]
-            res = torch._foreach_abs(tensors1)
-            if (dtype is torch.float16 or dtype is torch.bfloat16) and TEST_WITH_ROCM:
-                self.assertEqual(res, expected, atol=1.e-3, rtol=self.dtype_precisions[dtype][0])
-
-                torch._foreach_abs_(tensors1)
-                self.assertEqual(res, tensors1)
+            tensors = op.sample_inputs(device, dtype, N)
+            ref_res = [op.ref(t) for t in tensors]
+
+            method = op.get_method()
+            inplace = op.get_inplace()
+            fe_res = method(tensors)
```

Comment:
nit: What does `fe` in `fe_res` stand for? Maybe just `expected_res` would be clearer